### PR TITLE
kbfs: 2.11.0 ->  4.1.0

### LINKED ISSUE #64248: Keybase out of date, KBFS severely out of date
## Issue description
Hi, I'm a Keybase developer and some of our users reported that the Nix packages were out of date.

Keybase 4.1.0 was released a while ago, so I believe these need to be updated: https://github.com/NixOS/nixpkgs/tree/master/pkgs/tools/security/keybase.

Additionally, the KBFS nix should point to the new remote, `github.com/keybase/client` with `subPackages = [ "go/kbfs/kbfsfuse" "go/kbfs/kbfsgit/git-remote-keybase" ]`, and be on the same version as the other packages (4.1.0).

Please let me know if you have any questions, happy to answer! Thanks.

### Steps to reproduce
N/A

## Technical details
N/A

### PR DESCRIPTION
**Pull request by worldofpeace**
The kbfs code has been moved to the same
repo as the keybase package.

###### Motivation for this change
Fixes #64248

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by das-g — APPROVED** [review]

**Comment by worldofpeace** [comment]
I will merge this once I've manually tested this on master.

File: pkgs/tools/security/keybase/kbfs.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "kbfs";
+  # must have the same version as the keybase package
+  version = keybase.version;
+
+  goPackagePath = "github.com/keybase/client";
+  subPackages = [ "go/kbfs/kbfsfuse" "go/kbfs/kbfsgit/git-remote-keybase" ];
+
+  dontRenameImports = true;
+
+  src = keybase.src;
```
Comment (kalbasit) [inline pkgs/tools/security/keybase/kbfs.nix]:
```suggestion
  inherit (keybase) src version;
```

And also remove version assignment above.

**Review by kalbasit — CHANGES_REQUESTED** [review]
No modules upstream? 😣

**Comment by worldofpeace** [comment]
Thanks @kalbasit.

> No modules upstream? :persevere:

Yeah unfortunately not. Seems that reorganization and things like that aren't a small task.

**Review by kalbasit — APPROVED** [review]

**Pull request merged**